client/src/main/java/org/ovirt/vdsm/jsonrpc/client/reactors/SSLEngineNioHelper.java

```diff
@@ -164,4 +176,38 @@ public void clearBuff() {
     public SSLEngine getSSLEngine() {
         return this.engine;
     }
+
+    public void verifyPeerCertificates() throws ClientConnectionException, CertificateException {
```

Comment:
I do not like this method. This code is asynchronous and what we attempt to do is to waiting on handshake to finish to validate peer certificates. I would rather see lines 196-201 added here: https://github.com/oVirt/vdsm-jsonrpc-java/blob/master/client/src/main/java/org/ovirt/vdsm/jsonrpc/client/reactors/SSLEngineNioHelper.java#L132

Reply:
To sum up our offline conversation.
- imho, blocking in this area does not make any harm because it's connection establishing phase and not the processing one.
- Proposed location for handling this cert validation would not work because it happens just too late ("process"). That means any ssl errors will get translated into  regular error responses (exactly as what happens now)
- another proposed location, 'OneTimeCallback' that is passed as argument to post-connect, is executed in the  "process" phase. Look above.
Currently I don't see how it could be implemented better without introducing more complex changes in both, vdsm json rpc client and engine's code.

Reply:
This is not the perfect solution but, without the better alternatives, we agreed with @pkliczewski that it can be merged. Resolving this conversation.